### ISSUE TITLE
poller_boost not finishing in time?

### ISSUE THREAD

**Issue by vKnmnn** [Cacti/cacti#3593]
I'm having an issue with my main Cacti Server.
Cacti is on version 1.2.12

After a few days of usage drawing graphs will be terribly slow and the boost table will grow indefinitely.
I've seen it at 90GB already.

I can manually issue poller_boost.php --force --verbose --debug and that will run for a while.
After it's finished, my graphs get drawn more quickly, but the poller_boost table will gradually grow again, until it gets slow

output of my last poller forced run
```
02.06.2020 10:30:23 - SYSTEM BOOST STATS: Time:3570,5000 RRDUpdates:1417934
02.06.2020 10:30:23 - SYSTEM BOOST DETAIL STATS: RRDUpdates:1417934, TotalTime:3571, get_records:14,08, results_cycle:134,34, rrd_filename_and_template:5,31, rrd_lastupdate:0,01, rrdupdate:124,87, delete:10,55
```
This is the overview page right now (03-06-2020)
![image](https://user-images.githubusercontent.com/16045883/83617082-e12a4300-a588-11ea-87d9-dbcbb3eb715e.png)

**Comment by vKnmnn** [comment]
Performance-Settings Tab:
![image](https://user-images.githubusercontent.com/16045883/83617885-03709080-a58a-11ea-95e9-f8b03965ee93.png)

**Comment by bmfmancini** [comment]
Hmm permissions ok ?
check your logs under stats and search for boost does it ever run aside from when you do it manually ?

**Comment by vKnmnn** [comment]
What permissions are you referring to, exactly? i'd like to make sure of that.

my logs show me this every two hours, but then at 10:46 UTC i started the poller manually, as you can see from the first screenshot. The frequency changes.
```
03.06.2020 00:42:05 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 02:42:06 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 04:42:08 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 06:47:12 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 08:47:16 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 10:47:29 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 12:47:29 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 14:52:29 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 15:32:17 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 15:37:20 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 15:42:21 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 15:47:19 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 15:52:13 - BOOST WARNING: Boost Poller terminated by user
03.06.2020 15:57:11 - BOOST WARNING: Boost Poller terminated by user

```

**Comment by vKnmnn** [comment]
regarding permissions issue:
in my limited understanding it should not be an issue, since running poller_boost from cli as www-data works fine.

**Comment by vKnmnn** [comment]
today, this again:
    
![image](https://user-images.githubusercontent.com/16045883/83736506-046cf500-a652-11ea-91dd-4dee98de5882.png)

**Comment by TheWitness** [comment]
You have to find out what is killing the processes.  Those error messages are coming from the Signal handler.  Cacti does not kill the boost process on it's own.  It issues a warning, but allows it to continue to process.  Maybe you have the suhosin php module installed?  If so, please uninstall it.

**Comment by vKnmnn** [comment]
i cannot find any hints of suhoshin module. However i found the phpsessionclean service that gets run every thirty minutes. I don't think that's the reason though, because boost poller gets killed every cycle. I run the cacti poller via systemd-timer. Do you think it's possible that systemd kills all (forked) child-processes when the parent exits?


```

08.06.2020 14:02:39 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:07:32 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:12:36 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:18:14 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:22:39 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:27:35 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:32:49 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:37:24 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:42:37 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:47:39 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:52:29 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 14:57:41 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 15:03:08 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 15:07:42 - BOOST WARNING: Boost Poller terminated by user
08.06.2020 15:12:39 - BOOST WARNING: Boost Poller terminated by user

```

**Comment by vKnmnn** [comment]
Yes, it seems that this is the case. poller.php starts various processes, among them poller_boost.php
when poller.php exits after spine has run, systemd then kills the remaining processes.

**Comment by vKnmnn** [comment]
Ok. i switched back to crontab.

forcing the boost_poller to run might have broken something in the DB. i get some of these messages now:
` CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '640-traffic_in-2020-06-04 21:40:33' for key 'PRIMARY'`

**Comment by netniV** [comment]
Do you have the stack trace that usually accompanies the error?

**Comment by bmfmancini** [comment]
I see that error once on a while the duplicate entry one is normally around
the time a device is deleted and the pollers sync ive never thought
anything of it

**Comment by vKnmnn** [comment]
```

08.06.2020  16:20:04 - CMDPHP SQL Backtrace:   (/poller_boost.php[137]:output_rrd_data(),  /poller_boost.php[277]:db_execute(),  /lib/database.php[213]:db_execute_prepared())
--
08.06.2020 16:20:04 - CMDPHP ERROR: A DB Exec  Failed!, Error: Duplicate entry '640-traffic_in-2020-06-04 21:40:33' for  key 'PRIMARY'

```

**Comment by TheWitness** [comment]
If you have remote data collectors, you might want to do a full sync.  That duplicate error means that you either have two data collectors collecting for the same device, or you've cloned your install at some point, but not updated a spine setting or something like that.  Could also be that you have multiple Cron entries in multiple places on the same server.  I'm not sure it could be anything else.

**Comment by vKnmnn** [comment]
i have disabled my remote pollers, and they accumulate zero devices for polling. I'm very sure that i only have one cron entry. That particular error went away with time, i do not know why.

My Boost-DB doesn't seem to be growing that much anymore, but in the utilities page i still cannot see, that it ever finished.
So something still is wrong

![image](https://user-images.githubusercontent.com/16045883/84357854-95018300-abc6-11ea-84c0-f7f710c876f0.png)

**Comment by bmfmancini** [comment]
So I see the following error 

2020/06/11 10:36:01 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '81' for key 'PRIMARY'
2020/06/11 10:26:18 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '79' for key 'PRIMARY'
2020/06/11 10:25:52 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '17' for key 'PRIMARY'

But again this was after some devices on a remote collecter were delected and a full sync was done then the errors show up

**Comment by TheWitness** [comment]
Something must be 'unusual' with your setup.  Even though the main cron process ends, processes spawned by them should be able to resume unless there is some tweak in systemd that is enforcing all sub-processes to remain in the cgroup and prevent escaping.

As such, please provide your Linux release and version.

**Comment by vKnmnn** [comment]
Yes, i get that feeling too. I'm experiencing so many issues on my cacti System. For sure there's something weird about it.
It's Ubuntu Server 18 though, so really nothing special. Updated packages and all.

The Cacti DB has been migrated from another install though. Could it be there are some lingering settings in the DB from back then?

For example, in the User Permissions tab i still see an Option about DPDiscover which used to be installed and now isn't anymore.

I also saw a setting for maximum runtime of boost. I set this to it's maximum of 90 minutes, because boost is supposed to run every 2h. Maybe it gets killed here because it is taking so long?

My DB sits on spinning metal if that's in any way important...

Hope that helps
